src/coreclr/vm/threadsuspend.cpp

```diff
@@ -1962,39 +1962,109 @@ void ThreadSuspend::UnlockThreadStore(BOOL bThreadDestroyed, ThreadSuspend::SUSP
 #endif
 }
 
+typedef BOOL(WINAPI* PINITIALIZECONTEXT2)(PVOID Buffer, DWORD ContextFlags, PCONTEXT* Context, PDWORD ContextLength, ULONG64 XStateCompactionMask);
+PINITIALIZECONTEXT2 pfnInitializeContext2 = NULL;
+
+#ifdef TARGET_X86
+#define CONTEXT_COMPLETE (CONTEXT_FULL | CONTEXT_FLOATING_POINT |       \
```

Comment:
A nit - you can use `CONTEXT_ALL | CONTEXT_EXCEPTION_REQUEST` for all targets

Reply:
Will change. Thanks!

Reply:
Turns out this is not a case on ARM64. 
On ARM64 `CONTEXT_ALL` also includes `x18`, which always contains TEB pointer in user mode. We are not getting it in `RedirectCurrentThreadAtHandledJITCase` case and that causes assertions, since we expect more.

Ill make this as it was originally.

Reply:
Hmm, interesting. On Unix, x18 is not handled in a special way and is always part of the context.